youtube_dl/extractor/pandora_tv.py

```diff
@@ -0,0 +1,40 @@
+# encoding: utf-8
+import re
+import json
+
+from .common import InfoExtractor
+from ..compat import (
+    compat_parse_qs,
+    compat_urllib_parse_urlparse,
+)
+from ..utils import (
+    find_xpath_attr,
+    unified_strdate,
+    get_element_by_attribute,
+    int_or_none,
+    qualities,
+)
+
+
+class PandoraTVIE(InfoExtractor):
+    _VALID_URL = r'http://channel.pandora.tv/channel/video.ptv\?ch_userid=(?P<userid>.*?)&prgid=(?P<prgid>.*?)&.*'
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        video_id = mobj.group('userid') + '_' + mobj.group('prgid')
```

Comment:
User id is not related to video id hence shouldn't be part of `video_id`.
You shouldn't rely on query arguments order, `http://channel.pandora.tv/channel/video.ptv?prgid=...&ch_userid=...` should be matched as well.